mobzy-systems/src/main/kotlin/com/mineinabyss/mobzy/systems/listeners/MobListener.kt

```diff
@@ -175,13 +174,147 @@ object MobListener : Listener {
+            if (tamable.isTamable && !tamable.isTamed && tamable.tameItem?.toItemStack() == itemInHand) {
+                val random = Random(1).nextDouble()
+                tamable.isTamed = true
+                tamable.owner = player.uniqueId
+                rightClicked.toGeary().getOrSetPersisting { tamable.isTamed; tamable.owner == player.uniqueId }
+                player.spawnParticle(
+                    Particle.HEART,
+                    rightClicked.location.apply { y += 1.5 },
+                    10,
+                    random,
+                    random,
+                    random
+                )
+                return
+            }
+
+            //TODO Heal mob if fed tameItem
+            if (tamable.isTamed && (mob.health != maxHealth) && itemInHand == tamable.tameItem?.toItemStack()) {
+                if (mob.health + 2 <= maxHealth!!) mob.health += 2 else mob.health = maxHealth
+                player.playSound(mob.location, Sound.ENTITY_HORSE_EAT, 1f, 1f)
+                player.spawnParticle(Particle.HEART, rightClicked.location.apply { y += 2 }, 4)
+            }
+            else if (tamable.isTamed && tamable.owner == player.uniqueId && itemInHand.type == Material.NAME_TAG) {
+                modelEntity.nametagHandler.setCustomName("nametag", itemInHand.itemMeta.displayName)
+                modelEntity.nametagHandler.setCustomNameVisibility("nametag", true)
+            }
+
+            else if (tamable.isTamed && !rideable.isSaddled && tamable.owner == player.uniqueId && player.isSneaking) {
+                val model = gearyEntity.get<ModelEngineComponent>() ?: return
+                val saddle = modelEntity.getActiveModel(model.modelId).getPartEntity("saddle")
+
+                if (saddle.isVisible && !rideable.isSaddled) saddle.setItemVisibility(false)
+                else saddle.setItemVisibility(true)
+            }
+
+            else if (tamable.isTamed && tamable.owner == player.uniqueId && rideable.isSaddled && player.isSneaking) {
+                val model = gearyEntity.get<ModelEngineComponent>() ?: return
+                val saddle = modelEntity.getActiveModel(model.modelId).getPartEntity("saddle")
+
+                rightClicked.toGeary().setPersisting(!rideable.isSaddled)
+                if (saddle.isVisible) saddle.setItemVisibility(rideable.isSaddled)
+            }
+        }
+    }
+
+    //TODO Find a way to render the lead that isnt scuffed
+    /** Handling leashing of entities with [ModelEngineComponent] */
+    @EventHandler
+    fun PlayerLeashEntityEvent.onLeashingMob() {
+        val gearyEntity = entity.toGearyOrNull() ?: return
+        val modelEntity = entity.toModelEntity() ?: return
+
+        gearyEntity.with { componentEntity: ModelEngineComponent ->
+            if (!componentEntity.leashable) return
+            broadcast(modelEntity.isInvisible)
+            modelEntity.isInvisible = false
+            modelEntity.entity.addPotionEffect(
```

Comment:
What does this actually do? Perhaps add a comment explaining

Reply:
The leash wont render unless it is tied to an actual entity
Which for ME is the BaseEntity. Sadly tho its very annoying to do as the mob will sometimes just not be invis
I know the way OR and other balloon cosmetics do this is by making a pufferfish mount it
its about the smallest entity, so it would be totally invisible essentially